vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -580,6 +580,21 @@ protected boolean isInputValuePresent() {
         return getElement().getProperty("_hasInputValue", false);
     }
 
+    @Override
+    public void setValue(LocalDate value) {
+        LocalDate oldValue = getValue();
+
+        super.setValue(value);
+
+        if (oldValue == null && value == null && isInputValuePresent()) {
+            // Clear the input element from possible bad input.
+            getElement().executeJs("this.inputElement.value = $0", "");
+            getElement().setProperty("_hasInputValue", false);
+            fireEvent(new ClientValidatedEvent(this, false,
+                    checkValidity(value).isError()));
```

Comment:
**note:** We should probably consider removing this `valid` argument at all. It has become useless since we rely on `isInputValuePresent()` when checking for bad input, instead of an `isClientValid`.

Reply:
At first glance, seems that `checkValidity(value).isError()` would always return `false` here. That would then set the `valid` state of the event to be `false` which sounds kind of sounds backwards. I tried with `!checkValidity(value).isError()` and tests seem to pass.

Reply:
Ah, you are right, it should be `!checkValidity(...` indeed. However, as I mentioned, this argument is unused. I have added it here only because the event is a public API and someone might subscribe to it.

Reply:
I guess I'll go open a PR removing it.

Reply:
> At first glance, seems that checkValidity(value).isError() would always return false here

By the way, it may return `true` if the component is `required`.

Reply:
Tried with this test case and the output is always the same, regardless if the date picker is required or not.

```java
@Test
public void foobar() {
    DatePicker picker = new DatePicker();
    picker.setRequired(true);
    picker.addClientValidatedEventListener(e -> {
        System.out.println("valid? " + e.isValid());
    });
    picker.getElement().setProperty("_hasInputValue", true);
    picker.clear();
}
```

Reply:
Created a [ticket](https://github.com/vaadin/flow-components/issues/4425) to deprecate and remove this argument. In the meanwhile, I just pass `true`.